package.json

```diff
@@ -173,6 +167,119 @@
           ],
           "default": "keep-up-to-date",
           "markdownDescription": "Only applicable with `#haskell.languageServerVariant#` set to `haskell-language-server`. Determine what to do when a new version of the language server is available."
+        },
+        "haskell.plugin.importLens.codeActionsOn": {
+          "scope": "resource",
+          "type": "boolean",
+          "default": true,
+          "description": "Enables explicit imports code actions"
+        },
+        "haskell.plugin.importLens.codeLensOn": {
+          "scope": "resource",
+          "type": "boolean",
+          "default": true,
+          "description": "Enables explicit imports code lenses"
+        },
+        "haskell.plugin.hlint.codeActionsOn": {
+          "scope": "resource",
+          "type": "boolean",
+          "default": true,
+          "description": "Enables hlint code actions (apply hints)"
+        },
+        "haskell.plugin.hlint.diagnosticsOn": {
+          "scope": "resource",
+          "type": "boolean",
+          "default": true,
+          "description": "Enables hlint diagnostics"
+        },
+        "haskell.plugin.eval.globalOn": {
+          "scope": "resource",
+          "type": "boolean",
+          "default": true,
+          "description": "Enables eval plugin"
+        },
+        "haskell.plugin.moduleName.globalOn": {
+          "scope": "resource",
+          "type": "boolean",
+          "default": true,
+          "description": "Enables module name plugin"
+        },
+        "haskell.plugin.splice.globalOn": {
+          "scope": "resource",
+          "type": "boolean",
+          "default": true,
+          "description": "Enables splice plugin (expand template haskell definitions)"
+        },
+        "haskell.plugin.haddockComments.globalOn": {
+          "scope": "resource",
+          "type": "boolean",
+          "default": true,
+          "description": "Enables haddock comments plugin"
+        },
+        "haskell.plugin.class.globalOn": {
+          "scope": "resource",
+          "type": "boolean",
+          "default": true,
+          "description": "Enables type class plugin"
+        },
+        "haskell.plugin.retrie.globalOn": {
+          "scope": "resource",
+          "type": "boolean",
+          "default": true,
+          "description": "Enables retrie plugin"
+        },
+        "haskell.plugin.tactic.globalOn": {
+          "scope": "resource",
+          "type": "boolean",
+          "default": true,
+          "description": "Enables wingman (tactic) plugin"
+        },
+        "haskell.plugin.tactic.config.max_use_ctor_actions": {
+          "title": "Max number of constructors",
+          "scope": "resource",
+          "type": "boolean",
+          "default": true,
+          "markdownDescription": "Maximum number of `Use constructor <x>` code actions that can appear"
+        },
```

Comment:
Shouldn't `max_use_ctor_actions` be an integer?
https://github.com/haskell/haskell-language-server/blob/86d67ffe28871a01f54204b38470bfb49507abc5/plugins/hls-tactics-plugin/src/Wingman/Types.hs#L82

Reply:
ugh, you are right, thanks for catching it, will fix it

Reply:
And `features` should be a string, instead of an object

Reply:
mm I use that one on purpose, to force vscode editor show a button where you can write arbitrary json (including strings I hope)

Reply:
But the server accepts only strings. Why don't we use a textbox?

Reply:
tbh i do not get fully that config option, but it seems an advanced one, even obfuscated and it seems to me that it could be a list of enumerations, no?
but if types say is a String, it is :slightly_smiling_face:

Reply:
It's a magic string that enables extra features of wingman: https://github.com/haskell/haskell-language-server/pull/1398 (though it can be obtained easily somehow :P)

Reply:
ok will change, if the config option gets another type, we can change here at that time